extract_endpoint/src/crypt_utils.py

```diff
@@ -0,0 +1,9 @@
+import hmac
+import typing
+
+
+def sign_string(salt: typing.Optional[str], key: str, data: str) -> str:
```

Comment:
if it's optional, it should default to `None`. This code would be simpler if it defaulted to `''` however, and was never `Optional[str]`